astropy/coordinates/tests/test_solar_system.py

```diff
@@ -503,58 +349,75 @@ def test_url_or_file_ephemeris(time):
     coord_by_filepath = get_body("earth", time, ephemeris=filepath)
 
     # Using the URL or filepath should give exactly the same results:
-    assert_quantity_allclose(coord_by_url.ra, coord_by_filepath.ra)
-    assert_quantity_allclose(coord_by_url.dec, coord_by_filepath.dec)
-    assert_quantity_allclose(coord_by_url.distance, coord_by_filepath.distance)
+    np.testing.assert_array_equal(coord_by_url.ra, coord_by_filepath.ra)
+    np.testing.assert_array_equal(coord_by_url.dec, coord_by_filepath.dec)
+    np.testing.assert_array_equal(coord_by_url.distance, coord_by_filepath.distance)
 
 
-@pytest.mark.remote_data
 @pytest.mark.skipif(not HAS_JPLEPHEM, reason="requires jplephem")
-def test_url_ephemeris_wrong_input():
-    time = Time("1960-01-12 00:00")
-    with pytest.raises((HTTPError, URLError)):
-        # A non-existent URL
+def test_ephemeris_non_existing_url(monkeypatch):
+    def request_invalid_url(*args, **kwargs):
+        raise HTTPError(code=404, msg="Not Found", fp=None, hdrs=None, url="")
+
+    monkeypatch.setattr("urllib.request.OpenerDirector.open", request_invalid_url)
```

Comment:
@StuartLittlefair , this monkeypatch makes call to `urllib.request.OpenerDirector.open` raise an error. I think this method is called internally by `get_body` (via `astropy.utils.data`).

Thanks for everyone for the work and the approval. Merging!